### ISSUE TITLE
Cover Set-position

### ISSUE THREAD

**Issue by davidrs16** [home-assistant/appdaemon#338]
Hi,

I´m trying hasdashboard with home assistant where I have my home covers add to it. The issue that I have is that hadasboard cover component doesn´t support set-coverposition and I cannot request an specific position from the hadashboard. Also I cannot see the actual position of the cover only if it is open or close.

Do you plan to add these feature to this cool dashboard?

Thanks!

**Comment by ReneTode** [comment]
it is hard to create something when you dont own the hardware.
what you can do is create a input_select in HA and an automation that sets the cover to the chosen select.
the input select can be used on the dashboard.

**Comment by davidrs16** [comment]
Thank you ReneTode, yes that what a workaround that I have been thinking. In any case I´m not using a propietary hardware for the covers, I use ESP8266  with MQTT to manage the covers and also the positions, inside home assistant I´m using the default cover object, and I manage from it the position.

Thank you!!

**Comment by ReneTode** [comment]
the advantage you get is that you dont see on or off but the position that the cover is in.
not all covers have such options and only have up and down. or on or off.

please close the issue if the answer was satisfying.

**Comment by Wihanh** [comment]
I have a kinda similar question, didn't really want to open a separate issue, but if I need to let me know.
I have a cover (garage door) it is set up as a cover in HA, but in reality is just a relay that pushes a button, as there is no way to tell the door to 'open' or 'close' but you can emulate someone pushing the button, which when closed will open and when open will close and when moving will stop.  I also have sensors that pick up when the door is completely closed, and if that sensor is not triggered the door is considered open. (no position)

The problem in HADashboard i have is that I have the 2 garage doors showing and showing closed correctly.  If i open or toggle via HomeAssistant the Dash updates correctly.  If i click on the dash, it toggles and the icon changes from closed to open, if i then click again to stop the door while it was opening, the icon changes to closed, even though HA still reports the door as open.  Refreshing the dash fixes this.  Pressing it again, changes the icon to closed straight away, rather than actually waiting for the door to close and then updating when HA updates.  So i guess the problem boils down to the fact that the icon changes when you click it, rather than waiting on HA to tell it, it is actually shut, or open.  Is there a parameter to change this behaviour?  I guess a similar parameter in HA is the optimistic flag on mqtt.

Cheers
Wihan

**Comment by ReneTode** [comment]
no there is no such thing.
the dashboard is optimistic by design.
it is on the list to create an option to make it pesimistic, but there is no timeframe when that is implemented.

it is possible to create a workaround, by creating an app that you trigger by an input_boolean and that checks the state and reports back. so then the state from the input_boolean would be resetted untill its actualy changed. but you need to do some programming for that.
